### ISSUE TITLE
[REFACTOR][Norm] restructure kernel and op directories for 8 norm operators

### ISSUE THREAD

**Issue by zhen8838** [tile-ai/TileOPs#393]
### Symptom / Motivation
The norm operator family is growing to 8 ops (16 kernels), but the current directory layout has inconsistencies:

- `layer_norm` and `rms_norm` kernels are flat files under `tileops/kernels/norm/`
- `batch_norm` kernels live in a separate top-level `tileops/kernels/batch_norm/` directory
- Op files are flat in `tileops/ops/` with no grouping by family

This layout does not scale for 8 ops and makes it harder to navigate and maintain the norm family as a cohesive unit.

### Root Cause Analysis
The original structure was designed when only `layer_norm` and `rms_norm` existed. `batch_norm` was added later as a separate top-level kernel directory. No per-op subdirectory convention was established.

### Related Files
**Kernels (current):**
- `tileops/kernels/norm/__init__.py`
- `tileops/kernels/norm/layer_norm.py` — `LayerNormKernel`
- `tileops/kernels/norm/rms_norm.py` — `RmsNormKernel`
- `tileops/kernels/batch_norm/` — `BatchNormFwdTrainKernel`, `BatchNormFwdInferKernel`, `BatchNormBwdKernel`

**Ops (current):**
- `tileops/ops/layer_norm.py` — `LayerNormOp`
- `tileops/ops/rms_norm.py` — `RmsNormOp`
- `tileops/ops/batch_norm.py` — `BatchNormFwdOp`, `BatchNormBwdOp`

**Tests & Benchmarks:**
- `tests/ops/test_layer_norm.py`, `test_rms_norm.py`, `test_batch_norm.py`
- `benchmarks/ops/bench_layer_norm.py`, `bench_rms_norm.py`, `bench_batch_norm.py`

## Goal
Reorganize kernels into per-op subdirectories under `tileops/kernels/norm/`, create `tileops/ops/norm/` subpackage, migrate existing code, scaffold empty directories for 5 future ops, and ensure all existing tests pass. No new kernel implementations.

**8 ops in scope:** `rms_norm`, `qk_norm`, `layer_norm`, `batch_norm`, `group_norm`, `instance_norm`, `ada_layer_norm`, `ada_layer_norm_zero`

**Excluded:** `fused_add_rmsnorm`, `fused_add_layer_norm`

### Target structure

```
tileops/kernels/norm/
├── __init__.py                      # re-exports all kernel classes
├── layer_norm/
│   ├── __init__.py                  # exports LayerNormKernel
│   └── fwd.py                      # migrated
├── rms_norm/
│   ├── __init__.py                  # exports RmsNormKernel
│   └── fwd.py                      # migrated
├── batch_norm/
│   ├── __init__.py                  # exports BatchNorm*Kernel
│   ├── fwd.py                      # migrated from kernels/batch_norm/
│   └── bwd.py                      # migrated from kernels/batch_norm/
├── qk_norm/__init__.py             # stub
├── group_norm/__init__.py          # stub
├── instance_norm/__init__.py       # stub
├── ada_layer_norm/__init__.py      # stub
└── ada_layer_norm_zero/__init__.py # stub

tileops/ops/norm/
├── __init__.py                      # re-exports all norm Op classes
├── layer_norm.py                    # migrated
├── rms_norm.py                      # migrated
└── batch_norm.py                    # migrated
```

### Steps

1. Create kernel subdirectories — move `layer_norm.py` → `layer_norm/fwd.py`, `rms_norm.py` → `rms_norm/fwd.py`, merge `kernels/batch_norm/` → `kernels/norm/batch_norm/`, create 5 stub `__init__.py` for future ops, rewrite `kernels/norm/__init__.py`
2. Create `tileops/ops/norm/` subpackage — migrate 3 op files with import adjustments (`from .op import Op` → `from ..op import Op`; batch_norm kernel path update)
3. Update `tileops/ops/__init__.py` — replace 3 flat imports with `from .norm import ...`
4. Update test and benchmark imports — 6 files change `tileops.ops.{op}` → `tileops.ops.norm.{op}`
5. Delete old files — remove flat kernel files, old `kernels/batch_norm/` dir, old flat op files

## Constraints
- No new kernel implementations — scaffolding and migration only
- `tileops.ops.LayerNormOp` (top-level import) must continue to work after migration
- `tileops.kernels.norm.LayerNormKernel` must continue to work after migration
- Avoid `torch.library.custom_op` duplicate registration during migration (do not import both old and new paths in the same process)

## Acceptance Criteria
- [ ] Modified files pass unit tests (`test_layer_norm`, `test_rms_norm`, `test_batch_norm`)
- [ ] `from tileops.kernels.norm import LayerNormKernel, RmsNormKernel, BatchNormFwdTrainKernel` works
- [ ] `from tileops.ops import LayerNormOp, RmsNormOp, BatchNormFwdOp, BatchNormBwdOp` works
- [ ] `from tileops.ops.norm import LayerNormOp, RmsNormOp, BatchNormFwdOp, BatchNormBwdOp` works
- [ ] Old files (`tileops/ops/layer_norm.py`, `tileops/kernels/batch_norm/`, etc.) are removed
- [ ] 5 stub directories exist for future ops (`qk_norm`, `group_norm`, `instance_norm`, `ada_layer_norm`, `ada_layer_norm_zero`)
- [ ] All `__init__.py` files have explicit `__all__` and `from .module import Symbol` re-exports
